shell/layouts/default.vue

```diff
@@ -199,175 +200,30 @@ export default {
-        this.getGroups();
-      }
-    },
-
-    namespaces(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    clusterReady(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    product(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    async currentProduct(a, b) {
-      if ( !isEqual(a, b) ) {
-        if (a.inStore !== b.inStore || a.inStore !== 'cluster' ) {
-          const route = {
-            name:   'c-cluster-product',
-            params: {
-              cluster: this.clusterId,
-              product: a.name,
-            }
-          };
-
-          await this.$store.dispatch('prefs/setLastVisited', route);
-        }
-      }
-    },
-
-    $route(a, b) {
-      this.$nextTick(() => this.syncNav());
-    },
-
-  },
-
-  async created() {
-    this.queueUpdate = debounce(this.getGroups, 500);
-
-    this.getGroups();
-
-    await this.$store.dispatch('prefs/setLastVisited', this.$route);
-  },
-
-  mounted() {
-    // Sync the navigation tree on fresh load
-    this.$nextTick(() => this.syncNav());
-  },
-
-  methods: {
-    async setClusterAsLastRoute() {
-      const route = {
-        name:   this.$route.name,
-        params: {
-          ...this.$route.params,
-          cluster: this.clusterId,
-        }
-      };
-
-      await this.$store.dispatch('prefs/setLastVisited', route);
-    },
-    handlePageAction(action) {
-      if (action.action === SET_LOGIN_ACTION) {
-        this.afterLoginRoute = this.getLoginRoute();
-        // Mark release notes as seen, so that the login route is honoured
-        markSeenReleaseNotes(this.$store);
-      }
-    },
-
-    getLoginRoute() {
-      return {
-        name:   this.$route.name,
-        params: this.$route.params
-      };
-    },
-
-    collapseAll() {
-      this.$refs.groups.forEach((grp) => {
-        grp.isExpanded = false;
-      });
+    currentType() {
+      return this.$route.params.resource || '';
     },
 
-    getGroups() {
-      if ( this.gettingGroups ) {
-        return;
-      }
-      this.gettingGroups = true;
-
-      if ( !this.clusterReady ) {
-        clear(this.groups);
-        this.gettingGroups = false;
-
-        return;
+    groups() {
+      if (!this.clusterReady || !this.allSchemas || !this.counts || !this.locale || !this.currentProduct) {
+        // Referencing these computed properties
+        // to trigger the nav tree to be recomputed if they change.
+        return [];
       }
-
       const clusterId = this.$store.getters['clusterId'];
       const currentProduct = this.$store.getters['productId'];
       const currentType = this.$route.params.resource || '';
       let namespaces = null;
```

Comment:
Would using the computed namespaces be more efficient as it uses the `activeNamespaceCache`? If so, the conditional and getter at L221-L225 could be removed as well.